### PR TITLE
Fixed subnet file upload to allow upload of different report types

### PR DESCRIPTION
**Pull request by jinapurapu**
Functions to process Health reports for Subnet upload were incorrectly included in the subnetUploadReq, preventing generalization for other reports. This PR breaks the processing steps out to allow the other reports to be uploaded.

### CODE REVIEW

File: pkg/subnet/utils.go [hunk 1]
```diff
@@ -74,16 +74,40 @@ func UploadAuthHeaders(apiKey string) map[string]string {
...
+	case "perf":
+		{
+			fmt.Println("perf upload goes here")
+			// placeholder for perf upload in switch
+		}
+	case "tenant_report":
+		{
+			fmt.Println("tenant report upload goes here")
+			// placeholder for perf upload in switch
+		}
```
Comment (bexsoft) [inline pkg/subnet/utils.go:96]: I think we can comment / remove these lines as for now no function is set

**Review by bexsoft — CHANGES_REQUESTED** [review]

**Review by bexsoft — COMMENTED** [review]
Code looks good so far, Just found a small issue trying to generate a health report after registering my cluster:
Screenshot 2023-05-26 at 19 57 38

In the terminal:

`E: 2023/05/27 01:58:09 ErrorWithContext:error on ReadMessage: read tcp [::1]:10000->[::1]:63855: use of closed network connection`

Not sure if issue is related to the change

File: pkg/subnet/utils.go [hunk 2]
```diff
@@ -74,16 +75,31 @@ func UploadAuthHeaders(apiKey string) map[string]string {
...
+	var infoBody []byte
+	var formDataType string
+	var e error
+	if uploadType == "health" {
+		infoBody, formDataType, e = processHealthReport(info)
+		if e != nil {
+			return nil, "", e
+		}
+	} else {
+		return nil, "", errors.New("Invalid Subnet upload type")
```
Comment (cesnietor) [inline pkg/subnet/utils.go:88]: could we make the error start with lowercase? Sharing some references: https://github.com/golang/go/wiki/Errors

File: pkg/subnet/utils.go [hunk 3]
```diff
@@ -74,16 +75,31 @@ func UploadAuthHeaders(apiKey string) map[string]string {
...
 }
 
-func UploadFileToSubnet(info interface{}, client *xhttp.Client, filename string, reqURL string, headers map[string]string) (string, error) {
-	req, e := subnetUploadReq(info, reqURL, filename)
+func ProcessUploadInfo(info interface{}, uploadType string) ([]byte, string, error) {
+	var infoBody []byte
+	var formDataType string
+	var e error
+	if uploadType == "health" {
+		infoBody, formDataType, e = processHealthReport(info)
```
Comment (cesnietor) [inline pkg/subnet/utils.go:83]: could we please rename the error variable to `err`? (also in other places in this doc) please see reference: https://github.com/golang/go/wiki/Errors#naming.

No need to define the variable at the top, we can do this by just defining them like `infoBody, formDataType, e :=...`

**Comment by jinapurapu** [comment]
> Code looks good so far, Just found a small issue trying to generate a health report after registering my cluster: Screenshot 2023-05-26 at 19 57 38
> 
> In the terminal:
> 
> `E: 2023/05/27 01:58:09 ErrorWithContext:error on ReadMessage: read tcp [::1]:10000->[::1]:63855: use of closed network connection`
> 
> Not sure if issue is related to the change

Replicated on baremetal, looking into it now

File: pkg/subnet/utils.go [hunk 4]
```diff
@@ -73,16 +74,28 @@ func UploadAuthHeaders(apiKey string) map[string]string {
...
 
-func UploadFileToSubnet(info interface{}, client *xhttp.Client, filename string, reqURL string, headers map[string]string) (string, error) {
-	req, e := subnetUploadReq(info, reqURL, filename)
+func ProcessUploadInfo(info interface{}, uploadType string) ([]byte, string, error) {
+	if uploadType == "health" {
+		infoBody, formDataType, err := processHealthReport(info)
+		if err != nil {
+			return nil, "", err
+		}
+		return infoBody, formDataType, nil
```
Comment (reivaj05) [inline pkg/subnet/utils.go:83]: It could just be `return processHealthReport(info)`

**Review by reivaj05 — APPROVED** [review]

**Comment by bexsoft** [comment]
Hi!, Is Upload issue resolved?

**Pull request closed without merging**